README.txt
===================
Twexter 0.4 demo
===================

Features
-------------------
- A very clean and simple usable interface.
- Minimal memory and process footprint.
- No server side requirements, 100% javascript, CSS and HTML.
- All twexter functions are encapsulated as a single javascript file.

Install
-------------------
Unpack files in twexter/ directory

Place directory in your web server's public directory.
In Debian/Ubuntu is: /var/www/

Visit the project's url
http://yourwebserver.com/twexter

Requirements
-------------------
Currently working with Gecko engine based browsers (like Firefox) only.
No server side requirements, any webserver will work.

Test latest code
-------------------
http://test.twext.com


Changelog
2008-06-12 05:54:43 +0000 Fixed title now updates correclty when editing a loaded twext
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@24 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:37 +0000 Title now shows when loading a saved twext file
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@23 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:35 +0000 Fixed untext function now correclty restores the text on the input boxes
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@22 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:32 +0000 Loading a saved file now shows in preview and on textboxes
Fixed parsing of JSON on loading function


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@21 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:29 +0000 Restored json2.js, confirmed this is a better library code
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@20 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:23 +0000 Ignoring all files in /saved/ directory from the repository
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@19 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:20 +0000 Replaced json2 with json.js library, it has better and latest code
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@18 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:17 +0000 Added all the loading capabilities to index file to load the twexted text as a javascript string and file
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@17 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:14 +0000 Added the loading function that will parse the loaded text
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@16 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:09 +0000 Added a basic loading interface implemented in php
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@15 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:54:01 +0000 Changed saving extension to .js instead of txt for easier loading
Changed redirection to load interface


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@14 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:57 +0000 Re implemented PHP as a glue language to perform loading tasks on index
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@13 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:54 +0000 Fixed margin on CSS to avoid reported bug
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@12 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:51 +0000 Added very basic save functionality via a small PHP script.
The saved song will be stored as a JSON string, in chunk pairs and with TXT extension.
Its very simple and can be improved a lot to suit user's needs.

You can only save a file once. No rewrites or editing allowed yet.


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@11 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:47 +0000 Added browser detection to force firefox only.
Changed interface according to specs on: http://twext.com/User_talk:Gabrielsaldana dated 080602


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@10 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:43 +0000 Added style controls functions to all controls.
Fixed styles not being inherited properly


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@9 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-12 05:53:38 +0000 Style controls now work for text and twxt colors
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@8 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-01 04:52:16 +0000 Merge branch 'simple'
Conflicts:

	index.php
	plugins/twexter/twexter.js
	plugins/twexter/xx.gif
	twexter/xx.gif
	xx.gif


git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@7 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-01 04:52:02 +0000 merged conflict
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@6 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-06-01 04:52:00 +0000 Adding index.html file...something went wrong..
git-svn-id: https://twexter.svn.sourceforge.net/svnroot/twexter@5 2f62a6f2-632e-0410-acc4-be7c51fbfb5f

2008-05-31 23:17:25 -0500 Changed README file, renamed gpl.txt to LICENSE.txt

2008-05-31 23:08:50 -0500 Removed all currently unused files

2008-05-31 23:06:07 -0500 Page now scrolls to top of window on twext preview update

2008-05-31 23:02:12 -0500 Added form tags and names to style controls

2008-05-31 22:47:02 -0500 Added HTML DocType declaration. Set to HTML 4.01 Strict.

2008-05-31 22:46:40 -0500 Added style controls

2008-05-31 22:45:26 -0500 Fixed uppercase title on twext preview

2008-05-31 22:45:05 -0500 Extracted twext styles from twexter interface styles in different CSS

2008-05-31 20:45:37 -0500 Changed the instructions

2008-05-30 22:34:19 -0500 Changed styles according to test.twext.com/simple

2008-05-30 22:09:29 -0500 Removed unnecesary code

2008-05-30 22:08:15 -0500 Added title to preview

2008-05-29 01:01:11 -0500 Fixed auto grow textareas
Now textareas grow correctly without reducing size when no scrolling was present.

2008-05-28 23:02:32 -0500 Added title textbox

2008-05-28 23:02:16 -0500 Adjusted spacing between lines

2008-05-28 00:32:46 -0500 Fixed dummy save to correctly store values in json.

2008-05-28 00:30:32 -0500 Added logo image

2008-05-28 00:23:38 -0500 Fixed "partially" the grow text areas function.
Still needs better algorithm. It is set to grow after a certain fixed height. It should do it under any condition.

2008-05-28 00:23:32 -0500 Added credits and better colors and styles

2008-05-27 23:30:46 -0500 Removed reference to load gtranslate api when not present.

2008-05-27 23:30:20 -0500 Added some text and styles for demo page presentation

2008-05-27 02:41:04 -0500 Cleaned google js from non gtranslate version.
Made textareas smaller


index.php
<!-- Twexter interface -->
<?
/**
 * The Twexter interface
 *
 * @author Gabriel Saldana <gsaldana@gmail.com>
 * @license GPL v2.0
 */
$script = 'saved/' . $_POST['script'];
$title = '' ;
if($_POST['script'])
    {
        $title = str_replace(".js",'',$_POST['script']);
        $title = str_replace('_', ' ',$title);
    }
if($script)
    {
        /* Read file contents and load as JSON string */
        $file = fopen($script, "r");
        $jscript = '';
        while(!feof($file))
            {
                // send the current file part to the browser
                $jscript .= fread($file, filesize($script));
            }
        fclose($file);
    }
?>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
  <head>
    <title>Twexter</title>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <link rel="stylesheet" href="twext.css" type="text/css" media="all">
    <link rel="stylesheet" href="print.css" type="text/css" media="print">
    <link rel="stylesheet" href="twexter.css" type="text/css" media="all">
<script src="browserdetect.js" language="javascript" type="text/javascript"></script>

<script>
BrowserDetect.init();
if((BrowserDetect.browser != "Firefox") && (BrowserDetect.browser != "Iceweasel"))
{
   alert("Please only test with Firefox.\n Get it at http://getfirefox.com");
   window.location = "http://getfirefox.com";
}
</script>
  </head>
  <body>
<h1 id="logo">
<a href="http://twext.com/faq">Twext</a>&nbsp;<img src="xx.gif" alt="Twexter">&nbsp;<a href="http://twext.com/method">Method</a>
</h1>
<div id="toplinks">
      <a href="load.php">FIND</a>
      <a href="javascript:toggle_stylecontrols('style-controls')" id="stylecontrols_button">STYLE</a>
      <a href="javascript:toggle_stylecontrols('twexter-interface')" id="stylecontrols_button">PRINT</a>
</div>
<h1 id="preview_title"><?= ($title) ? strtoupper($title) : '' ?></h1>
    <div id="preview"></div>
<div id="twexter-interface">
    <form action="save.php" method="post" accept-encoding="utf-8">
    <div id="titlediv">
      <label>Title/T&iacute;tulo </label>
      <input type="text" name="title" id="title" value="<?= strtoupper($title) ?>" />
    </div>
    <textarea name="text1" id="xscroll-left" onkeyup="javascript:growTextAreas('xscroll-left','xscroll-right')"></textarea>
    <textarea name="text2" id="xscroll-right"
  onkeyup="javascript:growTextAreas('xscroll-right','xscroll-left')"></textarea>
    <div id="buttons">
    <input type="button" value="twext me" onclick="javascript:twext_text();saveTwext(twext_parse(txtleft.value,txtright.value));"/>
      <input type="submit" value="save">
      <input type="hidden" name="twexted_text" id="forsave">
      </div>
      <!-- begin style controls -->
      <div id="style-controls" style="display:none">
	<form id="styles-form" action="javascript:void(0)" method="post">
	<table summary="twext styling controls">
	  <thead>
	  <tr>
	    <th>&nbsp;</th>
	    <th class="normal-control-title">Text</th>
	    <th>&nbsp;</th>
	    <th class="twext-control-title">Twext</th>
	  </tr>
	  </thead>
	  <tbody>
	  <tr>
	    <td class="normal-control">Color</td>
	    <td>
	      <select name="normal-color-control" onchange="javascript:setStyle('normal-color', this.value)">
		<option value="#000033">Default</option>
		<option value="blue">Blue</option>
		<option value="red">Red</option>
	      </select>
	    </td>
	    <td class="twext-control">color</td>
	    <td>
	      <select name="twext-color-control" onchange="javascript:setStyle('twxt-color', this.value)">
		<option value="#cccc99">Default</option>
		<option value="blue">Blue</option>
		<option value="red">Red</option>
	      </select>
	    </td>
	  </tr>
	  <tr>
	    <td class="normal-control">Font</td>
	    <td>
	      <select name="normal-font-control" onchange="javascript:setStyle('normal-font', this.value)">
		<option value="Verdana">Default</option>
		<option value="Arial">Arial</option>
		<option value="Times New Roman">Times</option>
	      </select>
	    </td>
	    <td class="twext-control">font</td>
	    <td>
	      <select name="twext-font-control" onchange="javascript:setStyle('twxt-font', this.value)">
		<option value="Verdana">Default</option>
		<option value="Arial">Arial</option>
		<option value="Times New Roman">Times</option>
	      </select>
	    </td>
	  </tr>
	  <tr>
	    <td class="normal-control">Bold</td>
	    <td>
	      <select name="normal-bold-control" onchange="javascript:setStyle('normal-bold', this.value)">
		<option value="bolder">Default</option>
		<option value="normal">Normal</option>
		<option value="bold">Bold</option>
	      </select>
	    </td>
	    <td class="twext-control">bold</td>
	    <td>
	      <select name="twext-bold-control" onchange="javascript:setStyle('twxt-bold', this.value)">
		<option value="normal">Default</option>
		<option value="bold">Bold</option>
		<option value="bolder">Bolder</option>
	      </select>
	    </td>
	  </tr>
	  <tr>
	    <td class="normal-control">Size</td>
	    <td>
	      <select name="normal-size-control" onchange="javascript:setStyle('normal-size', this.value)">
		<option value="12pt">Default</option>
		<option value="14pt">14</option>
		<option value="16pt">16</option>
	      </select>
	    </td>
	    <td class="twext-control">size</td>
	    <td>
	      <select name="twext-size-control" onchange="javascript:setStyle('twxt-size', this.value)">
		<option value="6pt">Default</option>
		<option value="8pt">8</option>
		<option value="10pt">10</option>
	      </select>
	    </td>
	  </tr>
	  <tr>
	    <td class="normal-control">Space</td>
	    <td>
	      <select name="normal-space-control" onchange="javascript:setStyle('normal-space', this.value)">
		<option value="26px">Default</option>
		<option value="15px">15px</option>
		<option value="32px">32px</option>
	      </select>
	    </td>
	    <td class="twext-control">space</td>
	    <td>
	      <select name="twext-space-control" onchange="javascript:setStyle('twxt-space', this.value)">
		<option value="-8px">Default</option>
		<option value="-5px">5px</option>
		<option value="-2px">2px</option>
	      </select>
	    </td>
	  </tr>
	  </tbody>
	</table>
	</form>
      </div>
</div>
      <!-- end style controls -->
<!--       <script type="text/javascript" src="http://www.google.com/jsapi"></script> -->
      <script src="json2.js" language="javascript" type="text/javascript"></script>
      <script src="twexter.js" language="javascript"
      type="text/javascript"></script>
<script>
  function toggle_stylecontrols(controls)
{
  var ctls = document.getElementById(controls);
  if(ctls.style.display == 'none')
  {
   ctls.style.display = 'block';
  }
  else
  {
   ctls.style.display = 'none';
  }
}
</script>
    <? if(isset($jscript) && $jscript): ?>
     <script>
          var loadedtwext = "<?= $jscript ?>";
          loadTwext(loadedtwext);
     </script>
    <? endif ?>
   </body>
</html>
